sdk/identity/identity/src/credentials/interactiveBrowserCredential.ts

```diff
@@ -137,14 +137,22 @@ export class InteractiveBrowserCredential implements TokenCredential {
         try {
           const authResponse = await this.msalClient.acquireTokenByCode(tokenRequest);
           const successMessage = `Authentication Complete. You can close the browser and return to the application.`;
-          const expiresOnTimestamp = authResponse?.expiresOn.valueOf();
-          res.status(200).send(successMessage);
-          logger.getToken.info(formatSuccess(scopeArray));
-
-          resolve({
-            expiresOnTimestamp,
-            token: authResponse.accessToken
-          });
+          if (authResponse && authResponse.expiresOn) {
+            const expiresOnTimestamp = authResponse?.expiresOn.valueOf();
+            res.status(200).send(successMessage);
+            logger.getToken.info(formatSuccess(scopeArray));
+
+            resolve({
+              expiresOnTimestamp,
+              token: authResponse.accessToken
+            });
+          } else {
+            reject(
+              new Error(
+                `Device Authentication Error "Did not receive token with a valid expiration"`
```

Comment:
Pardon my ignorance, does the term "Device Authentication" apply to Interactive Browser credential as well? I assumed that it was specific to the Device Code Credential

Reply:
Good catch. Copy/paste issue